rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -0,0 +1,245 @@
+            if not stories:
+                return []
+
+            self._parse_stories(stories)
+            self._add_current_stories_to_result()
+            return self.story_steps
+
+        except ValueError:
+            logger.error(f"Failed to read {filename}, it will be skipped")
+
+        return []
+
+    def _parse_stories(self, stories: List[Dict[Text, Any]]) -> None:
+        for story_item in stories:
+            if not isinstance(story_item, dict):
+                raise_warning(
+                    f"Unexpected block found in {self.source_name}: \n"
+                    f"{story_item}\n"
+                    f"Items under the `{KEY_STORIES}` key must be YAML dictionaries."
+                    f"It will be skipped.",
+                    docs=DOCS_URL_STORIES,
+                )
+                continue
+
+            if KEY_PLAIN_STORY_NAME in story_item.keys():
+                self._parse_plain_story(story_item)
+
+    def _parse_plain_story(self, story_item: Dict[Text, Any]) -> None:
+        story_name = story_item.get(KEY_PLAIN_STORY_NAME, "")
+        if not story_name:
+            raise_warning(
+                f"Issue found in {self.source_name}: \n"
+                f"{story_item}\n"
+                f"The story has an empty name."
+                f"Stories should have a name defined under `{KEY_PLAIN_STORY_NAME}` key."
+                "It will be skipped.",
+                docs=DOCS_URL_STORIES,
+            )
+
+        steps = story_item.get(KEY_PLAIN_STORY_STEPS, "")
+
+        self._new_story_part(story_name, self.source_name)
+
+        for step in steps:
+            self._parse_step(step)
+
+    def _parse_step(self, step: Dict[Text, Any]) -> None:
+
+        if KEY_PLAIN_STORY_USER_UTTERANCE in step.keys():
+            self._parse_plain_story_user_utterance(step)
+        elif KEY_OR in step.keys():
+            self._parse_or_statement(step)
+        elif KEY_SLOT_NAME in step.keys():
+            self._parse_slot(step)
+        elif KEY_ACTION in step.keys():
+            self._parse_action(step)
+        elif KEY_CHECKPOINT in step.keys():
+            self._parse_checkpoint(step)
+        elif KEY_METADATA in step.keys():
+            pass
+        else:
+            raise_warning(
+                f"Issue found in {self.source_name}: \n"
+                f"Found an unexpected step in the story description:\n"
+                f"{step}\n"
+                "It will be skipped.",
+                docs=DOCS_URL_STORIES,
+            )
+
+    def _parse_plain_story_user_utterance(self, step: Dict[Text, Any]) -> None:
+
+        if self.use_e2e:
+            # TODO
+            pass
+        else:
+            utterance = self._parse_raw_user_utterance(step)
+            self.current_step_builder.add_user_messages([utterance])
+
+    def _parse_or_statement(self, step):
+        utterances = []
+
+        for sub_step in step.get(KEY_OR):
```

Comment:
what's a sub step? Should we keep closer to the wording which we use in the docs?

Reply:
each substep is an intent, right?

Reply:
I'd say it's `utterance`.

Reply:
then let's call it that?